crates/ink/codegen/src/generator/dispatch.rs

```diff
@@ -469,8 +480,68 @@ impl Dispatch<'_> {
                         const MUTATES: ::core::primitive::bool = #mutates;
                         const LABEL: &'static ::core::primitive::str = #label;
                         const ENCODING: ::ink::reflect::Encoding = ::ink::reflect::Encoding::Scale;
+                        }
+                    ))
+                }
+
+                if encoding.is_rlp() && !(encoding.is_all() && message.user_provided_selector().is_some()) {
```

Comment:
just a temporary fix for compilation. Added TODO in https://github.com/use-ink/ink/pull/2401